c/httpfileservice.c

```diff
@@ -197,6 +197,35 @@ void deleteUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
   }
 }
 
+void createFileFromUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
+  if (isDir(absolutePath)) {
+      int folderNameLen = strlen(absolutePath);
+      if(folderNameLen == 0 || absolutePath == NULL) {
+         respondWithJsonError(response, "Failed to idenity the folder pointed", 400, "Bad Request");
+      }
+      int slashPos = lastIndexOf(absolutePath, folderNameLen, '/');
+      char *tarFileName = (slashPos == -1) ? absolutePath : absolutePath + slashPos + 1;
+      char finalFileName[strlen(tarFileName) + strlen(".tar")];
+      strcpy(finalFileName, tarFileName);
+      strcat(finalFileName,".tar");
+      char finalCommand[strlen("/bin/sh -c 'tar -cf ") + strlen(finalFileName) + strlen(absolutePath) + 2];
+      strcpy(finalCommand, "/bin/sh -c 'tar -cf ");
+      strcat(finalCommand,finalFileName);
+      strcat(finalCommand," ");
+      strcat(finalCommand,absolutePath);
+      strcat(finalCommand,"'");
+      system(finalCommand);
+      if(doesFileExist(finalFileName)){
```

Comment:
Is it possible that:
* File already exists when you run the command
* The command fails (e.g. no write permissions to overwrite the file)
* You return "Success"

The caller will assume that the file has been successfully created whereas the file will still contain the old data.

Reply:
@ifakhrutdinov  so after serving the file we will delete it.

Reply:
so it will not exists anyways